src/components/RadioInput/RadioInput.js

```diff
@@ -0,0 +1,167 @@
+    content: '';
+    display: block;
+    position: absolute;
+    top: 50%;
+    left: ${theme.spacings.bit};
+    height: ${theme.spacings.byte};
+    width: ${theme.spacings.byte};
+    transform: translateY(-50%) scale(0, 0);
+    visibility: hidden;
+    transition: transform 0.05s ease-in, visibility 0.05s ease-in;
+  }
+`;
+
+const checkedStyles = ({ theme, checked }) =>
+  checked &&
+  css`
+    label: radio-input--active;
+
+    &::before {
+      border-color: ${theme.colors.b500};
+    }
+
+    &::after {
+      transform: translateY(-50%) scale(1, 1);
+      visibility: visible;
+    }
+  `;
+
+const invalidStyles = ({ theme, isInvalid }) =>
+  isInvalid &&
+  css`
+    label: radio-input--error;
+    &:not(:focus)::before {
+      background-color: ${theme.colors.r100};
+      border-color: ${theme.colors.r300};
+    }
+
+    &:not(:focus)::after {
+      background-color: ${theme.colors.r100};
+    }
+  `;
+
+const disabledStyles = ({ theme, disabled }) =>
+  disabled &&
+  css`
+    label: radio-input--disabled;
+    ${disableVisually()};
+
+    &::before {
+      border-color: ${theme.colors.n500};
+    }
+
+    &::after {
+      background-color: ${theme.colors.n500};
+    }
+  `;
+
+const inputStyles = ({ theme }) => css`
+  label: radio-input__input;
+  ${hideVisually()};
+
+  &:focus + label::before {
+    border-width: 2px;
+    border-color: ${theme.colors.b500};
+  }
+`;
+
+const Input = styled('input', { label: 'RadioInputInput' })(inputStyles);
+
+const Label = styled('label', { label: 'RadioInputLabel' })(
+  baseStyles,
+  checkedStyles,
+  disabledStyles,
+  invalidStyles
+);
+
+/**
+ * RadioInput component for forms.
+ */
+const RadioInput = ({ onClick, label, name, ...props }) => (
+  <Fragment>
+    <Input id={name} type="radio" {...{ ...props, name }} />
+    <Label htmlFor={name} {...{ ...props, onClick }}>
+      {label}
+    </Label>
+  </Fragment>
+);
+
+RadioInput.propTypes = {
+  /**
+   * Controles/Toggles the checked state.
+   */
+  onClick: PropTypes.func.isRequired,
+  /**
+   * Label string for this radio input.
+   */
+  label: PropTypes.string.isRequired,
+  /**
+   * Value string for input.
+   */
+  value: PropTypes.string.isRequired,
+  /**
+   * A unique ID used to link the input and label.
+   */
+  name: PropTypes.string.isRequired,
+  /**
+   * Triggers checked styles on the component. This is also forwarded as
+   * attribute to the <input> element.
+   */
+  checked: PropTypes.bool,
+  /**
+   * Triggers error styles on the component.
+   */
+  isInvalid: PropTypes.bool,
+  /**
+   * Triggers disabled styles on the component. This is also forwarded as
+   * attribute to the <input> element.
+   */
+  disabled: PropTypes.bool,
+  /**
+   * An ID rendered as data-selector attribute on the
+   * component. Used for tracking and e2e testing.
+   */
+  selector: PropTypes.string.isRequired
+};
+
+RadioInput.defaultProps = {
+  checked: false,
+  isInvalid: false,
+  disabled: false
```

Comment:
I find it a little funny to have `disabled` as boolean and then `isInvalid` right next to it. If we're going to use React final-form for most our projects, I'd say it makes sense to mirror their API (valid/invalid).

@felixjung - How did you do name this on the regular inputs?